x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformFailureHandlerTests.java

```diff
@@ -63,9 +64,145 @@ public int getFailureCountChangedCounter() {
         }
     }
 
-    public void testUnattended() {
+    // TODO: Circuit breaking exception
+
+    public void testHandleIndexerFailure_CircuitBreakingExceptionNewPageSizeLessThanMinimumPageSize() {
+        var e = new CircuitBreakingException(randomAlphaOfLength(10), 1, 0, randomFrom(CircuitBreaker.Durability.values()));
+
+        List.of(true, false).forEach((unattended) -> {
+            if (unattended) {
+                assertRetry(e, unattended);
+            } else {
+                assertFailure(e);
+            }
+        });
+    }
+
+    public void testHandleIndexerFailure_CircuitBreakingExceptionNewPageSizeNotLessThanMinimumPageSize() {
+        var e = new CircuitBreakingException(randomAlphaOfLength(10), 1, 1, randomFrom(CircuitBreaker.Durability.values()));
+
+        List.of(true, false).forEach((unattended) -> { assertNoFailureAndContextPageSizeSet(e, unattended, 365); });
+    }
+
+    public void testHandleIndexerFailure_ScriptException() {
+        var e = new ScriptException(
+            randomAlphaOfLength(10),
+            new ArithmeticException(randomAlphaOfLength(10)),
+            singletonList(randomAlphaOfLength(10)),
+            randomAlphaOfLength(10),
+            randomAlphaOfLength(10)
+        );
+        List.of(true, false).forEach((unattended) -> {
```

Comment:
nit: How about we move these to a function since we do this for multiple tests?

Reply:
Good point, I'll move this to a shared function.